application_form/models.py

```diff
@@ -9,14 +9,55 @@
     ("Muu", "Muu"),
 ]
 
-PERMISSIONS_LIST = [
-    ("add", _("Can add new applications.")),
-    ("change", _("Can change the existing applications.")),
-    ("delete", _("Can remove remove the existing applications.")),
+HASO_PERMISSIONS_LIST = [
+    ("haso_create", _("Can create new haso applications.")),
+    ("haso_update", _("Can update the existing haso applications.")),
+    ("haso_delete", _("Can remove remove the existing haso applications.")),
 ]
 
+HITAS_PERMISSIONS_LIST = [
+    ("hitas_create", _("Can create new hitas applications.")),
+    ("hitas_update", _("Can update the existing hitas applications.")),
+    ("hitas_delete", _("Can remove remove the existing hitas applications.")),
+]
+
+
+class ApplicationQuerySet(models.QuerySet):
+    def active(self):
+        return self.filter(
+            is_accepted=False, is_rejected=False, applicant_has_accepted_offer=False
+        )
+
+
+class ApplicationMixin(models.Model):
+    is_approved = models.BooleanField(default=False, verbose_name=_("is accepted"))
+    is_rejected = models.BooleanField(default=False, verbose_name=_("is rejected"))
+    rejection_description = models.TextField(
+        default="",
+        verbose_name=_("rejection description"),
+    )
+    applicant_has_accepted_offer = models.BooleanField(
+        default=False, verbose_name=_("applicant has accepted offer")
+    )
+
+    objects = ApplicationQuerySet.as_manager()
+
+    class Meta:
+        abstract = True
+
+    def save(self, **kwargs):
+        if self.is_approved and self.is_rejected:
+            raise ValueError(
+                _("application cannot be accepted and rejected at the same time.")
```

Comment:
Typo in word "application"

Reply:
Could you update this as well? 😄

Reply:
Hmm, I cannot see the typo here. So in the word "application"?

Reply:
Uhh, never mind. I think the word application just started to look incorrect even if it was written correctly. 👀